nncf/experimental/onnx/helper.py

```diff
@@ -0,0 +1,95 @@
+import os
+
+from typing import List
+
+import torchvision
+from torchvision import transforms
+
+import onnx
+
+from nncf.experimental.post_training.api.dataloader import DataLoader
+
+
+class ImageNetDataLoader(DataLoader):
```

Comment:
This shouldn't be part of NNCF. You are introducing a dependency of the ONNX path on the torch/torchvision, which the use may not want to have. Define a dataloader interface instead and program against it, letting the user pass the implementation they want. Consider using `nncf.common.initialization.dataloader.NNCFDataLoader`

Reply:
This is a helper for users that implements the interface of ImageNet torch Dataloader which is defined at `nncf.experimental.post_training.api.dataloader`
I see your point with the dependency on torch/torchvision. The solution could be to hide all imports under the functions/classes.

Reply:
Please, comment if you don't go along with the suggestion.